build_tools/install_mlir.sh

```diff
@@ -4,7 +4,11 @@
 # Usage (for aribtrary build/ directory):
 #   BUILD_DIR=/build ./build_tools/install_mlir.sh
 set -e
-td="$(realpath $(dirname $0)/..)"
+
+realpth() {
```

Comment:
Do you mind renaming this `portable_realpath`? (I can't look at this and not see a typo)

Also, in the past I've used something like this, which I feel like relies on less stackoverflow-bash-foo:

```
portable_realpath() {
  echo "$(cd $1 && pwd)"
}
```

And since there is only one usage here, I'd be open to just inlining it:

```
td="$(cd $(dirname $0)/.. && pwd)"
```

I don't mean to overly nit-pick such a basic script, but I feel like I've sprinkled `realpath` usage in a number of places, not knowing it was a GNU-ism, and would like to take this chance to confirm the right idiom that is portable so I can keep an eye out for it. Since I don't have a Mac, thanks for indulging me...

Reply:
Better to keep it a function so we don't inline every occurrence. LMK how this looks.